pkg/controller/operators/catalog/operator.go

```diff
@@ -697,11 +697,16 @@ func (o *Operator) syncSubscriptions(obj interface{}) error {
 		return fmt.Errorf("casting Subscription failed")
 	}
 
+	o.recordMetrics(sub)
```

Comment:
You may want to record metrics from the [subscriptionSyncer's `Sync` function](https://github.com/operator-framework/operator-lifecycle-manager/blob/e9ac62be99f467ccf201bd2af73cacea91ec714c/pkg/controller/operators/catalog/subscription/syncer.go#L46) since this sync function will only be called _after_ all others have successfully processed (i.e. you'll only get a fraction of total sync events).